airflow/api_fastapi/views/public/dags.py

```diff
@@ -71,16 +72,13 @@ async def get_dags(
 
     dags = session.scalars(dags_query).all()
 
-    try:
-        return DAGCollectionResponse(
-            dags=[DAGResponse.model_validate(dag, from_attributes=True) for dag in dags],
-            total_entries=total_entries,
-        )
-    except ValueError as e:
```

Comment:
Is `ValueError`Handled by FastAPI out-of-the-box and added to api specs?

Reply:
Good question. This was flask code, but basically we use a pydantic model to validate and convert data to a pydantic instance. Both the constructor and the `model_validate` only raise `ValidationError` is the data can't be validated.

Yes `ValidationError` will yield a 422 response and are nativelly handled by FastAPI so nothing more to do here.

That piece of code would need to remain if we could find a case that actually raises a ValueError, but I don't. And ValueError would raise a 500 internal error for the client. We don't have an exception handler for ValueError.